game-core/src/main/java/games/strategy/triplea/odds/calculator/ConcurrentBattleCalculator.java

```diff
@@ -69,20 +68,13 @@ public ConcurrentBattleCalculator(final String threadNamePrefix) {
     this.dataLoadedAction = dataLoadedAction;
   }
 
-  @Override
   public void setGameData(final GameData data) {
     synchronized (mutex) {
```

Comment:
Question/side-comment: if we are making a copy of `GameData` for the battle calculator, do you know why we need to lock it?

Reply:
I added a javadoc to the mutex field that should clarify this question.
Basically the mutex is not for thread safety, but rather for thread consistency.
Without this mutex it could happen that the gamedata is changed when we're actively deserializing the gamedata to create `BattleCalculator` workers, so the first x workers have the old data and the last y workers use the newer data which would screw up the results